target/linux/ramips/mt7621/base-files/lib/upgrade/dna.sh

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (C) 2023 Mauri Sandberg
+#
+
+# The vendor ubi is split in ubifs volumes 0-3 and the vendor root
+# filesystems are in volumes 2 (rootfs_0) and 3 (rootfs_1). Drop them and
+# explicitly use rootfs_0 as a boot partition that contains the dtb and the
+# OpenWrt kernel. This is because the u-boot expects to find them there.
+# Then continue upgrade with the default method and a squashfs rootfs will be
+# installed and the rest of ubifs will be used as overlay
+
+# The 'kernel' inside the sysupgrage.tar is an ubifs that contains /boot/dtb
+# and /boot/kernel. The 'root' is an OpenWrt squashfs root
+
+. /lib/functions.sh
+. /lib/upgrade/nand.sh
+
+dna_do_upgrade () {
```

Comment:
I am not an OpenWrt committer so take this with a grain of salt (especially if you do not agree): Other devices do not have their own sh-files in /lib/upgrade. So maybe the function can go in platform.sh?

Reply:
Under mt7621/base-files/lib/upgrade there are iodata.sh, ubnt.sh and zyxel.sh out of which at least the last one contains a function that is being used for one device only and iodata.sh. I'd rather leave it it's own file and not clutter the platform.sh especially if more er400 variants are to be added. We can change the filename to genexis.sh for example, to make it a bit more generic.

Reply:
Ok.